integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/InternalEventHandlerFactory.java

```diff
@@ -51,19 +53,25 @@
 class InternalEventHandlerFactory implements OpenLineageEventHandlerFactory {
 
   private final List<OpenLineageEventHandlerFactory> eventHandlerFactories;
-  private final VisitorFactory visitorFactory;
+  private final List<VisitorFactory> visitorFactory;
 
   public InternalEventHandlerFactory() {
     ServiceLoader<OpenLineageEventHandlerFactory> loader =
```

Comment:
The drawback of `ServiceLoader` approach is that it relies on `META-INF` files which can get lost whenever a user decides to build Uber-jar with his job packed together with `spark-openlinege`.  I would prefer using some other approach, like instantiating class by name.